src/burn/engine/search.cpp

```diff
@@ -1245,6 +1252,11 @@ static HRESULT MsiProductSearch(
     hr = BVariantChangeType(&value, type);
     ExitOnFailure(hr, "Failed to change value type.");
 
+    if (BURN_MSI_PRODUCT_SEARCH_TYPE_EXISTS == pSearch->MsiProductSearch.Type)
+    {
+        value.llValue = value.llValue == INSTALLSTATE_ABSENT ? 0 : 1;
```

Comment:
I'd like to see parens around the ternary condition:

```
value.llValue = (value.llValue == INSTALLSTATE_ABSENT) ? 0 : 1;
```

And moved up to the correct `case`. :)

Reply:
Isn't the value before BVariantChangeType string type?